### PR TITLE
fix: remove use of format in favour of inline bash

### LINKED ISSUE #1: filename_format_linux does not handle templating
Running on ubuntu-latest

```
  steps:
    - name: Install sccache
      uses: action-stars/install-tool-from-github-release@v0.2.0
      with:
        github_token: ${{ inputs.token }}
        owner: mozilla
        repository: sccache
        name: sccache
        extract: true
        version: ${{ inputs.version }}
        os_linux: unknown-linux-musl
        arch_amd64: x86_64
        subdirectory: "{name}-{version}-{arch}-{os}"
        filename_format_linux: "{name}-{version}-{arch}-{os}.{ext}"
        check_command: sccache --version
```

Note that the above _isn't strictly correct_ but is enough to illuminate the problem needs to really be : 
`"{name}-v{version}-{arch}-{os}"`

It causes this in the workflow-run logging
```
Run action-stars/install-tool-from-github-release@v0.2.0
  with:
    github_token: ***
    owner: mozilla
    repository: sccache
    name: sccache
    extract: true
    version: 0.7.0
    os_linux: unknown-linux-musl
    arch_amd64: x86_64
    subdirectory: {name}-{version}-{arch}-{os}
    filename_format_linux: {name}-{version}-{arch}-{os}.{ext}
    check_command: sccache --version
    tag_prefix: v
    arch_arm64: arm64
    os_macos: macos
    os_windows: windows
    extract_mode_linux: tar.gz
    extract_mode_macos: tar.gz
    extract_mode_windows: zip
    filename_format: {name}_{version}_{os}_{arch}.{ext}
Run action-stars/github-release-lookup@c86fa936d2dea43eac094b7d3e4ed486a93d1484
  with:
    github_token: ***
    owner: mozilla
    repository: sccache
    tag_prefix: v
    version: 0.7.0
Run ${GITHUB_ACTION_PATH}/scripts/lookup.sh
  ${GITHUB_ACTION_PATH}/scripts/lookup.sh
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    GITHUB_TOKEN: ***
    OWNER: mozilla
    REPOSITORY: sccache
    VERSION: 0.7.0
    TAG_PREFIX: v
Error: The template is not valid. action-stars/install-tool-from-github-release/v0.2.0/action.yml (Line: 109, Col: 20): The following format string is invalid: [https://github.com/${{](https://github.com/$%7B%7B) inputs.owner }}/${{ inputs.repository }}/releases/download/${{ inputs.tag_prefix }}{version}/%s
```

Looks like https://github.com/action-stars/install-tool-from-github-release/blob/58e2dd20166c0eb19ab9ac4d0966c930a647ee69/action.yml#L109C56-L111 doesn't like the extra templating `{}` that is being used. (no doubt because it's nested go templating).

- The downstream _pbrisbin/setup-tool-action_ doesn't say whether templating is supported or not when you specify the linux_url_. 
- It does work if you switch to using `filename_format`, because you don't use `format()`

### PR DESCRIPTION
**Pull request by quotidian-ennui**
# Motivation

Resolves #1

## Changes

- Use inline bash to derive the supported platform_urls
- Fixed a minor copy & paste error around platform descs

Test run : https://github.com/quotidian-ennui/install-tool-from-github-release/actions/runs/6772382912

### CODE REVIEW

**Comment by quotidian-ennui** [comment]
@stevehipwell  -> Steve, you might want to have a CODEOWNERS or similar to assign yourself to review this ;)

File: action.yml [hunk 1]
```diff
@@ -91,6 +91,19 @@ runs:
...
+      run: |
+        if [[ -n "${{ inputs.filename_format_linux }}" ]]; then
+          echo "linux_url=https://github.com/${{ inputs.owner }}/${{ inputs.repository }}/releases/download/${{ inputs.tag_prefix }}{version}/${{ inputs.filename_format_linux }}" >> "$GITHUB_OUTPUT"
+        fi
+        if [[ -n "${{ inputs.filename_format_macos }}" ]]; then
+          echo "macos_url=https://github.com/${{ inputs.owner }}/${{ inputs.repository }}/releases/download/${{ inputs.tag_prefix }}{version}/${{ inputs.filename_format_macos }}" >> "$GITHUB_OUTPUT"
+        fi
+        if [[ -n "${{ inputs.filename_format_windows }}" ]]; then
+          echo "windows_url=https://github.com/${{ inputs.owner }}/${{ inputs.repository }}/releases/download/${{ inputs.tag_prefix }}{version}/${{ inputs.filename_format_windows }}" >> "$GITHUB_OUTPUT"
+        fi
```
Comment (stevehipwell) [inline action.yml:106]:
```suggestion
        set -euo pipefail
        
        if [[ -n "${{ inputs.filename_format_linux }}" ]]
        then
          echo "linux_url=https://github.com/${{ inputs.owner }}/${{ inputs.repository }}/releases/download/${{ inputs.tag_prefix }}{version}/${{ inputs.filename_format_linux }}" >> "${GITHUB_OUTPUT}"
        fi
        
        if [[ -n "${{ inputs.filename_format_macos }}" ]]
        then
          echo "macos_url=https://github.com/${{ inputs.owner }}/${{ inputs.repository }}/releases/download/${{ inputs.tag_prefix }}{version}/${{ inputs.filename_format_macos }}" >> "${GITHUB_OUTPUT}"
        fi
        
        if [[ -n "${{ inputs.filename_format_windows }}" ]]
        then
          echo "windows_url=https://github.com/${{ inputs.owner }}/${{ inputs.repository }}/releases/download/${{ inputs.tag_prefix }}{version}/${{ inputs.filename_format_windows }}" >> "${GITHUB_OUTPUT}"
        fi
```

**Review by stevehipwell — CHANGES_REQUESTED** [review]
I've added a _nit_ comment for the bash script. Could you also split the PR in half so we can get the action changes merged ASAP.

RE the tests I think a matrix job driven from a YAML file processed in a pre-job would be the best way to make this extensible.

**Comment by quotidian-ennui** [comment]
Ok then, I committed your suggestions, and I removed the test-install-tool yaml; I'll raise a new PR with a matrix style.

**Review by stevehipwell — APPROVED** [review]
Thanks @quotidian-ennui!

**Pull request merged**